x-pack/plugins/security_solution/public/flyout/right/hooks/use_highlighted_fields.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { TimelineEventsDetailsItem } from '@kbn/timelines-plugin/common';
+import { find, isEmpty } from 'lodash/fp';
+import { ALERT_RULE_TYPE } from '@kbn/rule-data-utils';
+import { isAlertFromEndpointEvent } from '../../../common/utils/endpoint_alert_check';
+import {
+  getEventCategoriesFromData,
+  getEventFieldsToDisplay,
+} from '../../../common/components/event_details/get_alert_summary_rows';
+
+export interface UseHighlightedFieldsParams {
+  /**
+   * An array of field objects with category and value
+   */
+  dataFormattedForFieldBrowser: TimelineEventsDetailsItem[] | null;
+}
+
+export interface UseHighlightedFieldsResult {
+  /**
+   * Highlighted field name (label or if null, falls back to id)
+   */
+  field: string;
+  description: {
+    /**
+     * Highlighted field name (overrideField or if null, falls back to id)
+     */
+    field: string;
+    /**
+     * Highlighted field value
+     */
+    values: string[] | null | undefined;
```

Comment:
curious why `values` here cannot be stricter. in line 78 and 79, if item.values is falsy, it will not be added to result?

Reply:
I agree, but typescript is complaining about it... the only way I found was to add a `ts-ignore` for the reduce, but I didn't like that...